docs/notary-server.md

```diff
@@ -0,0 +1,154 @@
+and signing a timestamp file for each repo whenever a client sends updates,
+after verifying the root, target, and snapshot signatures on the client update.
+
+### Authentication
+
+Notary Server supports authentication from clients using [JWT](http://jwt.io/)
+tokens.  This requires an authorization server that manages access controls,
+and a cert bundle from this authorization server containing the public key it
+uses to sign tokens.
+
+The client will log into the server (it gets redirected by Notary Server if
+authentication is configured), obtain a token, and present the token to Notary
+Server, which should be configured to trust signatures from that authorization
+server.
+
+See the docs for [Docker Registry v2 authentication](
+https://github.com/docker/distribution/blob/master/docs/spec/auth/token.md)
+for more information.
+
+### Server storage
+
+Currently Notary Server uses MySQL as a backend for storing the timestamp
+*public* keys and the TUF metadata for each repository.
+
+### Signing service
+
+The recommended usage of the server is with a separate signing service:
+[Notary Signer](notary-signer.md).  The signing service actually
+stores the timestamp *private* keys and performs signing for the server.
+
+By using a signing service, the private keys then would never be stored on the
+server itself.
+
+Notary Signer supports mutual authentication - when you generate client
+certificates for Notary Server to authenticate with Notary Signer, please make
+sure that the certificates **are not CAs**.  Otherwise any server that is
+compromised can sign any number of other client certs.
+
+As an example, please see [this script](opensslGenCert.sh) to see how to
+generate client SSL certs with basic constraints using OpenSSL.
+
+### How to configure and run notary server
+
+A JSON configuration file is used to configure Notary Server.  Please see the
+[Notary Server configuration document](notary-server-config.md)
+for more details about the format of the configuration file.
+
+The parameters of the configuration file can also be overwritten using
+environment variables of the form `NOTARY_SERVER_var`, where `var` is the
+full path from the top level of the configuration file to the variable you want
+to override, in all caps.  A change in level is denoted with a `_`.
+
+For instance, one part of the configuration file might look like:
+
+```json
+"storage": {
+	"backend": "mysql",
+	"db_url": "dockercondemo:dockercondemo@tcp(notary-mysql)/dockercondemo"
+}
+```
+
+If you would like to specify a different `db_url`, the full path from the top
+of the configuration tree is `storage -> db_url`, so the environment variable
+to set would be `NOTARY_SERVER_STORAGE_DB_URL`.
+
+Note that you cannot override an intermediate level name.  Setting
+`NOTARY_SERVER_STORAGE=""` will not disable the MySQL storage.  Each leaf
+parameter value must be set indepedently.
+
+#### Running a Docker image
```

Comment:
Note: I'm writing about this as if the official docker image already exists - not sure what the name will be, so this may need to be fixed.